tests/director/tests/bundle_instance_auth_per_scenario_test.go

```diff
@@ -0,0 +1,290 @@
+		Description: ptr.String("my first wordpress application"),
+		Labels:      graphql.Labels{ScenariosLabel: []interface{}{scenario}},
+	}
+
+	appInputGQL, err := testctx.Tc.Graphqlizer.ApplicationRegisterInputToGQL(appInput)
+	require.NoError(t, err)
+
+	request := fixtures.FixRegisterApplicationRequest(appInputGQL)
+	actualApp := graphql.ApplicationExt{}
+
+	err = testctx.Tc.RunOperation(ctx, dexGraphQLClient, request, &actualApp)
+	cleanups = append([]func(){func() {
+		//fixtures.DeleteApplicationLabel(t, ctx, dexGraphQLClient, actualApp.ID, ScenariosLabel)
+		fixtures.SetApplicationLabel(t, ctx, dexGraphQLClient, actualApp.ID, ScenariosLabel, []interface{}{conf.DefaultScenario})
+		fixtures.UnregisterApplication(t, ctx, dexGraphQLClient, defaultTenant, actualApp.ID)
+	}}, cleanups...)
+	if !assert.NoError(t, err) {
+		return nil, executeCleanUp, false
+	}
+
+	//Register Bundle
+	bndlName := "test-bundle" + uniqueStr
+	authInput := fixtures.FixBasicAuth(t)
+
+	bndlInput := fixtures.FixBundleCreateInputWithDefaultAuth(bndlName, authInput)
+	bndl, err := testctx.Tc.Graphqlizer.BundleCreateInputToGQL(bndlInput)
+	require.NoError(t, err)
+
+	addBndlRequest := fixtures.FixAddBundleRequest(actualApp.ID, bndl)
+	bndlAddOutput := graphql.Bundle{}
+
+	err = testctx.Tc.RunOperation(ctx, dexGraphQLClient, addBndlRequest, &bndlAddOutput)
+	cleanups = append([]func(){func() { fixtures.DeleteBundle(t, ctx, dexGraphQLClient, defaultTenant, bndlAddOutput.ID) }}, cleanups...)
+	if !assert.NoError(t, err) {
+		return nil, executeCleanUp, false
+	}
+
+	// Request bundle instance auth
+	rtmAuth := fixtures.RequestClientCredentialsForRuntime(t, context.Background(), dexGraphQLClient, defaultTenant, actualRuntime.ID)
+	rtmOauthCredentialData, ok := rtmAuth.Auth.Credential.(*graphql.OAuthCredentialData)
+	require.True(t, ok)
+	require.NotEmpty(t, rtmOauthCredentialData.ClientSecret)
+	require.NotEmpty(t, rtmOauthCredentialData.ClientID)
+
+	accessToken := token.GetAccessToken(t, rtmOauthCredentialData, token.RuntimeScopes)
+	oauthGraphQLClient := gql.NewAuthorizedGraphQLClientWithCustomURL(accessToken, conf.GatewayOauth)
+
+	authCtx, inputParams := fixtures.FixBundleInstanceAuthContextAndInputParams(t)
+	bndlInstanceAuthRequestInput := fixtures.FixBundleInstanceAuthRequestInput(authCtx, inputParams)
+	bndlInstanceAuthRequestInputStr, err := testctx.Tc.Graphqlizer.BundleInstanceAuthRequestInputToGQL(bndlInstanceAuthRequestInput)
+	require.NoError(t, err)
+
+	bndlInstanceAuthCreationRequestReq := fixtures.FixRequestBundleInstanceAuthCreationRequest(bndlAddOutput.ID, bndlInstanceAuthRequestInputStr)
+	actualBundleInstanceAuth := graphql.BundleInstanceAuth{}
+
+	err = testctx.Tc.RunOperation(ctx, oauthGraphQLClient, bndlInstanceAuthCreationRequestReq, &actualBundleInstanceAuth)
+	cleanups = append([]func(){func() {
+		fixtures.DeleteBundleInstanceAuth(t, ctx, dexGraphQLClient, defaultTenant, actualBundleInstanceAuth.ID)
+	}}, cleanups...)
```

Comment:
No needed as this will be deleted as part of application's deletion.

Reply:
The deferred operation (remove application from the formation) cannot proceed otherwise